pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/MetadataStoreFactoryImpl.java

```diff
@@ -78,6 +123,8 @@ public static String removeIdentifierFromMetadataURL(String metadataURL) {
             return metadataURL.substring(EtcdMetadataStore.ETCD_SCHEME_IDENTIFIER.length());
         } else if (metadataURL.startsWith(ZKMetadataStore.ZK_SCHEME_IDENTIFIER)) {
             return metadataURL.substring(ZKMetadataStore.ZK_SCHEME_IDENTIFIER.length());
+        } else if ((provider = findProvider(metadataURL)) != null) {
```

Comment:
We could also have the `MetadataStoreProvider` for the existing implementations and in `loadProviders()` we always go through the list of builtin providers as well what is specified in the property.